include/fem-dem/cfd_dem_coupling.h

```diff
@@ -237,13 +237,56 @@ class CFDDEMSolver : public GLSVANSSolver<dim>
   dynamic_flow_control() override;
 
   /**
-   * @brief Check if the disabling contacts is enabled and that
+   * @brief contact_search_step
+   * Checks all the conditions that require a contact search step. The check of
+   * conditions is done in order of suspected frequency occurrence.
    *
    */
+
   inline bool
-  contacts_are_disabled(unsigned int counter) const
+  contact_search_step(const unsigned int counter)
   {
-    return has_disabled_contacts && counter > 1;
+    if (contact_detection_step)
```

Comment:
Thanks for the comments, damn

Reply:
What do you mean haha

Reply:
your comments are clear. In general, your code is commented in a very elegant way. You are very good at this, that's all I mean. I was like "Damn this is good commenting"

Reply:
Ohhh, thanks you haha